src/server/requestHandlers.js

```diff
@@ -105,36 +105,28 @@ fluid.defaults("sjrk.storyTelling.server.getStoryHandler", {
     }
 });
 
-/**
+/**
  * Gets a single story's data from the database, parses it, rebuilds any URLs to
  * files associated with that story and responds to the HTTP request
  *
  * @param {Object} request - a Kettle request that includes an ID for the story to retrieve
  * @param {Component} dataSource - an instance of sjrk.storyTelling.server.dataSource.couch.story
- * @param {String} uploadedFilesHandlerPath - the path on the server to uploaded files
  */
-sjrk.storyTelling.server.handleGetStory = function (request, dataSource, uploadedFilesHandlerPath) {
+sjrk.storyTelling.server.handleGetStory = function (request, dataSource) {
     var id = request.req.params.id;
     var promise = dataSource.get({directStoryId: id});
 
     promise.then(function (response) {
+        if (response.published) {
 
-        fluid.transform(response.content, function (block) {
-            if (block.blockType === "image") {
-                if (block.imageUrl) {
-                    block.imageUrl = uploadedFilesHandlerPath + "/" + block.imageUrl;
-                }
-                return block;
-            } else if (block.blockType === "audio" || block.blockType === "video") {
-                if (block.mediaUrl) {
-                    block.mediaUrl = uploadedFilesHandlerPath + "/" + block.mediaUrl;
-                }
-                return block;
-            }
-        });
-
-        var responseAsJSON = JSON.stringify(response);
-        request.events.onSuccess.fire(responseAsJSON);
+            var responseAsJSON = JSON.stringify(response);
+            request.events.onSuccess.fire(responseAsJSON);
+        } else {
+            request.events.onError.fire({
+                isError: true,
+                message: "An error occurred while retrieving the requested story"

```

Comment:
I've been going back and forth on this one, not sure whether I should be worried about enumeration attacks. Since we're using v4 UUID's, it's not impossible to guess ID's, but it's going to be very difficult and not a very practical vector, IMO.

If I'd like to avoid giving away that a story ID was correct but that it's unpublished, then it would be better to match this error message to the one below it. I'll make them the same and send the actual error message on the server log.